cli/cmd/inject.go

```diff
@@ -158,29 +130,22 @@ func injectDaemonSet(bytes []byte) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	podTemplateSpec := injectPodTemplateSpec(&ds.Spec.Template)
-	if podTemplateSpec == nil {
+	if !injectPodTemplateSpec(&ds.Spec.Template) {
 		return nil, nil
 	}
-	return enhancedDaemonSet{
-		&ds,
-		enhancedDaemonSetSpec{
-			&ds.Spec,
-			*podTemplateSpec,
-		},
-	}, nil
+	return &ds, nil
 }
 
 /* Given a PodTemplateSpec, return a new PodTemplateSpec with the sidecar
  * and init-container injected. If the pod is unsuitable for having them
  * injected, return null.
  */
-func injectPodTemplateSpec(t *v1.PodTemplateSpec) *enhancedPodTemplateSpec {
+func injectPodTemplateSpec(t *v1.PodTemplateSpec) bool {
```

Comment:
I am curious to know what your motivation was behind changing the return type of `injectPodTemplateSpec`. From what I understand, it looks like we are changing this function so it modifies object instead of return a newly generated one. Is that right?

Reply:
> I am curious to know what your motivation was behind changing the return type of injectPodTemplateSpec.

First, I *had* to change the return type since I am removing `enhancedPodTemplateSpec`.

> From what I understand, it looks like we are changing this function so it modifies object instead of return a newly generated one. Is that right?

Before this PR, the code was modifying the original object too. This PR changes things to just modify it a little more.

The second reason I changed the return type is that it was confusing to return a "new" object that had its internals aliasing the old option. It's surprising that a function modifies its input function when it returns a replacement object.

Reply:
Ahh that makes sense. I didn't realize enhancedPodTemplateSpec was an entirely new return type that was been returned. And I agree, it can be a little confusing if a newly returned object still returns old parameters. Having a "modification" function rather than a "generator" function sounds like a good approach.